pkg/storage/mongo/connect_util.go

```diff
@@ -0,0 +1,47 @@
+package mongo
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/globalsign/mgo"
+)
+
+// ConnDetails is a convenient way to store connection details for a mongo
+// server
+type ConnDetails struct {
+	Host     string
+	Port     int
+	User     string
+	Password string
+	Timeout  time.Duration
+	SSL      bool
+}
+
+func (c ConnDetails) String() string {
+	return fmt.Sprintf(`Host: %s, Port: %d, User: %s, Password: <redacted>, Timeout: %s`, c.Host, c.Port, c.User, c.Timeout)
+}
+
+// GetSession is a utility to create a session from a DB connection string
+// and credentials. The caller is responsible for calling Close on the returned
+// session.
+//
+// This function always establishes connections via TLS
```

Comment:
This may no longer be true with the environment variable `MONGO_SSL` which could be false.

Reply:
good catch, thanks @robjloranger